### ISSUE TITLE
Pick cheaper defaults: This CF script will cost ~$750 USD per month to run with default instance sizes on US-West

### ISSUE THREAD

**Issue by willcodeforfood** [aws-quickstart/quickstart-magento#23]
I installed Magento 2 on my AWS account using this template, and literally overnight, I owed over $100 USD for it.

When deployed on US-West, the estimated cost was about $750 USD per month. The main offenders look to be the Aurora DB (`db.r3.2xlarge`) and Elasticache Redis (`cache.m4.large`).

Luckily, it's an easy fix -- just change the instance sizes to make them smaller. Downgrading to `cache.t2.micro` and `db.t2.small` brings the costs down to a much more manageable level, and doesn't seem to impact the performance of the Magento 2 store (at least, for someone playing with it and developing -- obviously not for a production store).

I think it would be best to start with small instances as defaults, then let users bump up the size as they see fit.

**Comment by santiagocardenas** [comment]
Absolutely, we should make those instance classes selectable, so a user can modify before deployment. We are open to taking pull requests if you are inclined to help with that change. It mostly required parameterizing those and creating parameters with allowed values in all respective templates similar to the instance type parameters.

**Comment by sammarcus** [comment]
(@willcodeforfood by the way, if you haven't already emailed billing support, they're usually pretty reasonable about this and will reverse the usage charges if you explain what happened.)

**Comment by mathro** [comment]
Guys, please fix this. This is burning money for no reason.

**Comment by ray-moncada** [comment]
if you use cache.t2.micro make sure to disable the automatic fail over feature, T1 and T2 do not support automatic fail over feature.

**Comment by ray-moncada** [comment]
> I installed Magento 2 on my AWS account using this template, and literally overnight, I owed over $100 USD for it.
> 
> When deployed on US-West, the estimated cost was about $750 USD per month. The main offenders look to be the Aurora DB (`db.r3.2xlarge`) and Elasticache Redis (`cache.m4.large`).
> 
> Luckily, it's an easy fix -- just change the instance sizes to make them smaller. Downgrading to `cache.t2.micro` and `db.t2.small` brings the costs down to a much more manageable level, and doesn't seem to impact the performance of the Magento 2 store (at least, for someone playing with it and developing -- obviously not for a production store).
> 
> I think it would be best to start with small instances as defaults, then let users bump up the size as they see fit.

if you use cache.t2.micro make sure to disable the automatic fail over feature, T1 and T2 do not support automatic fail over feature.

**Comment by yehuda-elementryx** [comment]
@willcodeforfood do you recall what this stack cost to run after to downgrading to `cache.t2.micro` and `db.t2.small` ?

**Comment by ray-moncada** [comment]
Sorry, I do not. One more thing I like to point out is that composer does not run under micro.

**Comment by willcodeforfood** [comment]
I did not leave it running for very long, and I'm not sure how to decipher my bill from that month tbh. I've attached a screenshot of that month's bill to help you get an idea 
![image](https://user-images.githubusercontent.com/1639118/53267775-ddcbba80-36a9-11e9-9ea2-f7ce006041bc.png)

Disclaimer: I did have a few other things running in my AWS account that month, but all the big expensive instances are from this Magento 2 CF script.
